specs/experimental/dispute-game-v2/dispute-game-v2.md

```diff
@@ -0,0 +1,519 @@
+    }
+
+    /// @notice Enter a subcontext with a disputed node and an agreed upon node.
+    /// @param `subcontextType`     - The type of subcontext to enter.
+    /// @param `challengedBranchId` - The ID of the branch node being challenged. The
+    ///                               disputed node entered into the subcontext will
+    ///                               always the the leaf within the branch.
+    function enterSubcontext(
+      SubcontextType subcontextType,
+      uint256 challengedBranchId
+    ) external payable;
+
+    /// @notice Attempts to resolve a subgame.
+    /// @param `branchId` - The ID of the branch node subgame being resolved.
+    function resolveSubgame(uint256 branchId) external {
+        // ...
+    }
+
+    /// @notice Claims all available credit for the given recipient.
+    /// @param `recipient` - The recipient to claim credit for.
+    function claimCredit(address recipient) external {
+        // ...
+    }
+}
+```
+
+### Aggregate Dispute Game
+
+The Aggregate Dispute Game is a new high-level `IDisputeGame` that implements the `BisectionGame` abstract contract.
+At a high level, the purpose of the aggregate game is to allow for re-use of an output bisection component, spawning
+various [subcontexts](#isubcontext) to validate the claims made on a single block state transition.
+
+The structure of this game, now that it will use the `BisectionGame` primitive, is such that we can add additional
+layers as time goes on, such as during the [interop hardfork](../../interop/overview.md).
+
+#### Output Bisection (Top Layer)
+
+The first layer inside of the Aggregate Dispute Game is the "output bisection" component. This will be the
+highest-level dispute, implementing `IDisputeGame` and created by the `DisputeGameFactory`.
+
+In this `BisectionGame` implementation, the `enterSubcontext` function shall be implemented with several available
+"verifier subcontexts" (i.e., `FaultBisectionGame`, `ZKVerifier`, etc.)
+
+#### `FaultBisectionGame`
+
+The `FaultBisectionGame` is a second-level subcontext, enabling further bisection over a fault proof program instruction
+trace between two blocks. In this `BisectionGame` implementation, the `enterSubcontext` function simply passes the
+arguments to the FPVM contract's `step` function to determine validity at the leaf level.
+
+#### `ZKVerifier`
+
+The `ZKVerifier` is a second-level subcontext, enabling atomic verification of a validity proof for the single-block
+state transition.
+
+## Security Considerations
+
+### Challenger DoS
+
+Because of the low-cost to challenge a valid game and require a verification subcontext to be entered, the challenger
+is opened up to DoS attack. However, assuming the cost of proving off-chain outweighs the best-case bond, the system
+is still economically sound.
+
+## Credits
+
+- Daejun Park, Matt Gleason, and Mofi Taiwo for the initial ideas of branch claims.
+
+[pow]: https://ethresear.ch/t/fraud-proofs-are-broken/19234
+
+<style>
```

Comment:
This renders literally in the github markdown, not sure if it applies or not, maybe in the mdbook?

Reply:
Yeah in the mdbook it renders as a style tag to center the mermaid graphs.